packages/client/hmi-client/src/components/code/tera-code.vue

```diff
@@ -268,6 +297,14 @@ const fileNames = computed<string[]>(() => {
 	return Object.keys(codeAsset.value?.files);
 });
 
+const codeAssetCopy = ref<Code | null>(null);
```

Comment:
this is a copy of the original asset, so that when changes are made and we cancel them we can use the default asset data